src/net.cpp

```diff
@@ -944,12 +946,11 @@ void ThreadSocketHandler2(void* parg)
                 if (nErr != WSAEWOULDBLOCK)
                     LogPrintf("socket error accept INVALID_SOCKET: %d", nErr);
             }
-            else if (nInbound >= gArgs.GetArg("-maxconnections", 125) - MAX_OUTBOUND_CONNECTIONS)
+            else if (nInbound >= max_connections - MAX_OUTBOUND_CONNECTIONS)
             {
                 LogPrint(BCLog::LogFlags::NET,
-                         "Surpassed max inbound connections maxconnections:%" PRId64 " minus max_outbound:%i",
-                         gArgs.GetArg("-maxconnections", 125),
-                         MAX_OUTBOUND_CONNECTIONS);
+                         "Surpassed max inbound connections of %i",
+                         std::max<int>(max_connections - MAX_OUTBOUND_CONNECTIONS, 0));
```

Comment:
this can display 0 when it shouldn't.  if max outbound is 8 and max connectionx is 5 it would show 0 since 5-8 is -3

I really dislike the wording of this.

I feel we could just say something like "Surpassed maximum connection count; new inbound connection has been dropped" or declined and show the max_connections and if u wanted a more debug level you could add the max_connections and max_outbound_connections.

Reply:
Why don’t you put up a commit to fix it.

Reply:
its more of a nit and we can do that change later on. I approve of everything done to prevent crashes